### DISCUSSION TITLE
How to get wallet name after connection

### DISCUSSION THREAD

**Discussion by nicolas-zozol** [WalletConnect/walletconnect-monorepo#3501] [Developer Support] [answered]
Some of our users to connect with WalletConnect protocol, with Safe Wallet, Argent and others
I would like to know where I can find the information of the wallet ? I can ifnd it in the dashboard, so it is somewhere :)

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/912634/af74007d-6492-416f-b0ba-4c3b00a5ceb7)

**Comment by glitch-txs** [comment]
Hi which package are you using? Usually is in `provider.session.peer.name` I think

**Comment by nicolas-zozol** [comment, marked as answer]
I'm using web3js with Blocknative Onboard, and it's on `web3.provider.connector.session.peer.metadata.name`
Thank you, it saved me at least 3 days of work 🥰


![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/912634/33e79aab-fb33-4751-bb70-08ca5884fba1)
